uportal-war/src/main/java/org/jasig/portal/rest/PeopleRESTControllerV43.java

```diff
@@ -0,0 +1,124 @@
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.servlet.ModelAndView;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+@Controller
+@RequestMapping("/v4-3/people/")
+public class PeopleRESTControllerV43 {
+
+    private IPersonManager personManager;
+    private ObjectMapper jsonMapper;
+
+    @Autowired
+    public void setJsonMapper(ObjectMapper jsonMapper) {
+        this.jsonMapper = jsonMapper;
+    }
+
+    @Autowired
+    public void setPersonManager(IPersonManager personManager) {
+        this.personManager = personManager;
+    }
+
+    private PersonLookupHelperImpl lookupHelper;
+
+    @Autowired(required = true)
+    public void setPersonLookupHelper(PersonLookupHelperImpl lookupHelper) {
+        this.lookupHelper = lookupHelper;
+    }
+
+    @RequestMapping(value="/", method = RequestMethod.GET)
+    public void searchPeople(@RequestParam Map<String,Object> query,
+            HttpServletRequest request, HttpServletResponse response) throws IOException {
+
+        final IPerson user = personManager.getPerson((HttpServletRequest) request);
+        if (user == null) {
+            response.setStatus(HttpServletResponse.SC_UNAUTHORIZED);
+            return;
+        }
+
+        List<IPersonAttributes> people = lookupHelper.searchForPeople(user, query);
+
+        if ( people == null ) {
+            response.setStatus(HttpServletResponse.SC_NOT_FOUND);
+            return;
+        }
+
+        //Unwrap attributes
+        List<Object> results = new ArrayList<Object>();
+        for(IPersonAttributes p : people) {
+            results.add(p.getAttributes());
+        }
+
+        response.setHeader("Content-Type", "application/json");
+        jsonMapper.writeValue(response.getOutputStream(), results);
+    }
+
+    @RequestMapping(value="/{username}", method = RequestMethod.GET)
+    public ModelAndView getPerson(@PathVariable String username,
+            HttpServletRequest request, HttpServletResponse response) {
+
+        final IPerson searcher = personManager.getPerson(request);
+        if (searcher == null) {
+            response.setStatus(HttpServletResponse.SC_UNAUTHORIZED);
+            return null;
+        }
+
+        final IPersonAttributes person = lookupHelper.findPerson(searcher, username);
+        if (person == null) {
+            response.setStatus(HttpServletResponse.SC_NOT_FOUND);
+            return null;
+        }
+
+        return new ModelAndView("json", person.getAttributes());
+    }
+
+    @RequestMapping(value = "/me", method = RequestMethod.GET)
```

Comment:
I don't find this argument compelling...

> It would break the conventions in the rest of this controller; if I see /me.json and try /andrewstuart.json, I'm going to get a 404.

because (1) the controller is entirely new with this commit (and we have the opportunity simply to add `.json` across the board);  and (2) omitting the `.json` is actually a "break" from the pattern we've been following with REST APIs.

BUT!  I do think the first point holds water, and (inspired by this discussion) I've been doing some googling & reading on the topic of "restful API naming conventions" and "rest api naming best practices."  My research suggests we're in the minority... perhaps an extremely slim minority.. when we add `.json` to our URIs.

When we began that practice, I think there was a real sense that we -- and other typical API designers -- would commonly offer a "/people/drewwills.json" side-by-side with a "/people/drewwills.xml."  That hasn't panned out in uPortal, or the industry more broadly.

So I feel I could easily be persuaded to flip this rule;  but I'd like for more folks to chime in, and -- if we change -- I'd believe we should **communicate** and **document** this choice effectively.